### ISSUE TITLE
[test-runner-visual-regression] Add selenium launcher and SauceLabs support

### ISSUE THREAD

**Issue by web-padawan** [modernweb-dev/web#806]
At Vaadin we use SauceLabs for visual regression tests, so I would like to add `selenium` support.

It seems like this might need the following changes:

1. Expose `driver` publicly and use [WebDriver.findElement](https://github.com/SeleniumHQ/selenium/blob/63d986984f9c5d3bd78f41ef16ff7664cac8763b/javascript/node/selenium-webdriver/lib/webdriver.js#L950):

```js
await driver.findElement(); // add locator function
```

2. Take a screenshot using [WebElement.takeScreenshot](https://github.com/SeleniumHQ/selenium/blob/63d986984f9c5d3bd78f41ef16ff7664cac8763b/javascript/node/selenium-webdriver/lib/webdriver.js#L2316):

```js
const screenshot = await element.takeScreenshot();
```

So I tried to make these changes but apparently `IFrameManager` makes it more complicated than I though.

I'm going to continue investigating how to do it, but would be very grateful for a piece of advice 🙂

**Comment by LarsDenBakker** [comment]
In selenium we execute the tests all in a single page using iframes, that proved to be a lot faster and stable in the end.

This definitely creates an issue here in the end... some thoughts.

1. Finding the correct element
Here we need to know which iframe we're currently executing in. Ideally we can reference a session id to the correct iframe.

2. Taking the screenshot
I'm not sure what happens when you try to screenshot an individual element in an iframe. All the iframes are positioned on top of each other taking the full width of the page. We could perhaps hide the other iframes while taking a screenshot, but I don't know if that will interfere with the actual tests.

**Comment by web-padawan** [comment]
Thanks for the info. Would it be possible to only use one iframe when using visual regression plugin?

**Comment by LarsDenBakker** [comment]
A concurrency of 1 should accomplish that.

out of curiousity, did you use visual regression testing on saucelabs with another tools? seems pretty tricky to get stable/fast.

**Comment by web-padawan** [comment]
We use `gemini` and another framework by the same team,`hermione`. But running the tests there is very slow, 8-10 seconds per HTML suite, so I'd like to switch to WTR. AFAIK both tools use a fork of `webdriverio`.

**Comment by web-padawan** [comment]
Fixed by #1054
